src/parser/shaman/enhancement/modules/talents/LightningShield.tsx

```diff
@@ -7,106 +7,94 @@ import Analyzer, { SELECTED_PLAYER } from 'parser/core/Analyzer';
-      Events.applybuff.by(SELECTED_PLAYER)
-        .spell(SPELLS.LIGHTNING_SHIELD_TALENT),
-      this.onApplyBuff,
+      this.onLightningShieldDamage,
     );
   }
 
-  onDamage(event: DamageEvent) {
-    this.damage += event.amount + (event.absorbed || 0);
-  }
-
-  onApplyBuff() {
-    this.overchargeCount += 1;
-  }
-
-  get damagePercent() {
-    return this.owner.getPercentageOfTotalDamageDone(this.damage);
+  onLightningShieldDamage(event: DamageEvent) {
+    this.damageGained += event.amount + (event.absorbed || 0);
   }
 
-  get damagePerSecond() {
-    return this.damage / (this.owner.fightDuration / 1000);
-  }
-
-
   get getLightningShieldUptime() {
-    return this.selectedCombatant.getBuffUptime(SPELLS.LIGHTNING_SHIELD_TALENT.id) / this.owner.fightDuration;
+    return this.selectedCombatant.getBuffUptime(SPELLS.LIGHTNING_SHIELD_TALENT.id) / (this.owner.fightDuration - this.deathTracker.totalTimeDead);
   }
 
   get suggestionThresholds() {
     return {
       actual: this.getLightningShieldUptime,
       isLessThan: {
-        major: 1,
+        minor: 1,
+        major: 0.95,
+        recommended: 1.00,
       },
       style: 'percentage',
     };
   }
 
   suggestions(when: any) {
     when(this.suggestionThresholds)
-      .addSuggestion((suggest: any, actual: any) => {
+      .addSuggestion((suggest: any, actual: any, recommended: any) => {
         return suggest(
           <Trans>
-            You should fully utilize your <SpellLink id={SPELLS.LIGHTNING_SHIELD_TALENT.id} /> by using it before combat.
+            Your <SpellLink id={SPELLS.LIGHTNING_SHIELD_TALENT.id} /> uptime can be improved.
+            You can benefit from the full passive damage increase by casting it before combat.
+            {this.deathTracker.totalTimeDead ? <>Your time spent death has already been deducted from the required uptime.</> : ''}
           </Trans>,
         )
           .icon(SPELLS.LIGHTNING_SHIELD_TALENT.icon)
-          .actual(
-            <Trans>
-              You kept up <SpellLink id={SPELLS.LIGHTNING_SHIELD_TALENT.id} /> for ${formatPercentage(actual)}% of the fight.</Trans>,
+          .actual(<Trans>{formatPercentage(actual)}% uptime</Trans>,
           )
-          .recommended(<>It is possible to keep up <SpellLink id={SPELLS.LIGHTNING_SHIELD_TALENT.id} /> for 100% of the fight by casting it pre-combat.</>);
+          .recommended(<Trans>{formatPercentage(recommended)}% is recommended.</Trans>);
       });
   }
 
   statistic() {
     return (
       <Statistic
-        position={STATISTIC_ORDER.OPTIONAL()}
+        position={STATISTIC_ORDER.CORE()}
         size="small"
         category={'TALENTS'}
       >
-        <BoringSpellValueText
-          spell={SPELLS.LIGHTNING_SHIELD_TALENT}
-        >
-          <>
-            <ItemDamageDone amount={this.damage} />
-          </>
+        <BoringSpellValueText spell={SPELLS.LIGHTNING_SHIELD_TALENT}>
+            <ItemDamageDone amount={this.damageGained} />
```

Comment:
Personally i would consider including other information like the Uptime considering it is a threshold for a suggestion. 

Also idk if you are using ItemDamageDone because you think you have to or because you like the layout of it, but you can lay it out pretty much however you want, as an example, you could do something like this....

```
<BoringSpellValueText spell={SPELLS.LIGHTNING_SHIELD_TALENT}>
    <>
        <ItemDamageDone amount={this.damageGained} /><br />   
        <UptimeIcon /> {formatPercentage(this.getLightningShieldUptime,0)}% <small>buff uptime</small>
    </>
</BoringSpellValueText>
```

The end result would look like this (if you do this, make sure you change the size of the stat box to "flex" instead of small
![image](https://user-images.githubusercontent.com/1304554/82852059-5fbb1c80-9ec7-11ea-8345-bcd122a964d9.png)

Or you could display the raw damage done instead of DPS, or you could display both, or you could display the important stuff and put the rest in a tooltip ... pretty much whatever you want. There are plenty of examples around the other specs.

Reply:
Ignore me, i didnt know that lightning shield was a 1 hour buff. Disgreard